app/api/utility.py

```diff
@@ -55,16 +65,29 @@ def create_query(
     """
     subject_level_filters = ""
 
-    if isinstance(age, tuple):
-        if age[0] is not None:
-            subject_level_filters += "\n" + f"FILTER (?{AGE.var} >= {age[0]})."
-        if age[1] is not None:
-            subject_level_filters += "\n" + f"FILTER (?{AGE.var} <= {age[1]})."
+    if age[0] is not None:
+        subject_level_filters += "\n" + f"FILTER (?{AGE.var} >= {age[0]})."
+    if age[1] is not None:
+        subject_level_filters += "\n" + f"FILTER (?{AGE.var} <= {age[1]})."
 
     if sex is not None:
-        # select_str += f' ?{GENDER_VAR}'
         subject_level_filters += "\n" + f"FILTER (?{SEX.var} = '{sex}')."
 
+    if diagnosis is not None:
+        subject_level_filters += (
+            "\n" + f"FILTER (?{DIAGNOSIS.var} = {diagnosis})."
+        )
+
+    if is_control is not None:
+        if is_control:
+            subject_level_filters += (
+                "\n" + f"FILTER (?{IS_CONTROL.var} = <{IS_CONTROL_TERM}>)."
+            )
+        else:
+            subject_level_filters += (
```

Comment:
oh, interesting. I hadn't even thought of that. This is the "give me only patients" use case? Good idea.

This also looks good to me, I'll just leave a note that [SPARQL has a negation pattern](https://www.w3.org/TR/sparql11-query/#negation) for the `FILTER` command and I am not completely sure if the two are equivalent when there is more than one relation of the same type (e.g. if you somehow end up with several `bg:isSubjectGroup` statements, one of which meets your Filter condition here.

We should make sure we understand what the differences are between the two.

Reply:
Interesting, thanks for sharing! I've made a note of this.